src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php

```diff
@@ -231,16 +231,38 @@ public function testEncodeRootAttributes()
         $this->assertEquals($expected, $this->encoder->encode($array, 'xml'));
     }
 
-    public function testEncodeCdataWrapping()
+    /**
+     * @dataProvider encodeCdataWrappingWithDefaultPattern
+     */
+    public function testEncodeCdataWrappingWithDefaultPattern($input, $expected)
     {
-        $array = [
-            'firstname' => 'Paul & Martha <or Me>',
+        $this->assertEquals($expected, $this->encoder->encode($input, 'xml'));
+    }
+
+    public static function encodeCdataWrappingWithDefaultPattern()
+    {
+        return [
+            [['firstname' => 'Paul and Martha'], '<?xml version="1.0"?>'."\n".'<response><firstname>Paul and Martha</firstname></response>'."\n"],
```

Comment:
```suggestion
    public static function encodeCdataWrappingWithDefaultPattern(): iterable
    {
        yield [
            ['firstname' => 'Paul and Martha'],
            '<?xml version="1.0"?>'."\n".'<response><firstname>Paul and Martha</firstname></response>'."\n",
        ];
```

Same for the others please